### PR TITLE
Sync recent problem spec changes

### LINKED ISSUE #230: flatten-array: add additional instructions about Tcl's lack of NULL value.
Good to have here - if it's not yet in the problem prompt (not included in this diff), it could be added there also.

_Originally posted by @hiljusti in https://github.com/exercism/tcl/pull/229#discussion_r780562867_

### PR DESCRIPTION
**Pull request by glennj**
Tests for:
- accumulate
- allergies
- bowling
- flatten-array; add extra instructions
- forth
- high-scores
- largest-series-product
- phone-number
- rational-numbers
- word-count
- word-search

Docs for a few other exercises.

One commit for each exercise

Fixes #230

### CODE REVIEW

**Review by kotp — APPROVED** [review]

File: exercises/practice/flatten-array/flatten-array.test [hunk 1]
```diff
@@ -1,8 +1,11 @@
-#!/usr/bin/env tclsh
+##!/usr/bin/env tclsh
```
Comment (angelikatyborska) [inline exercises/practice/flatten-array/flatten-array.test:1]: What is this change about?

File: exercises/practice/forth/forth.test [hunk 2]
```diff
@@ -11,234 +14,253 @@ test forth-1 "numbers just get pushed onto the stack" -body {
 } -returnCodes ok -result {1 2 3 4 5}
 
 skip forth-2
-test forth-2 "can add two numbers" -body {
+test forth-2 "numbers just get pushed onto the stack" -body {
+    evalForth "-1 -2 -3 -4 -5"
+} -returnCodes ok -result {-1 -2 -3 -4 -5}
```
Comment (angelikatyborska) [inline exercises/practice/forth/forth.test:19]: The description of this test was supposed to be `"pushes negative numbers onto the stack"` according to the problem specs. I'm not sure if this matters. In some languages and their testing libraries, test descriptions must be unique.

**Review by angelikatyborska — COMMENTED** [review]
I reviewed all changes the best I could without actually knowing tcl 😇

File: exercises/practice/flatten-array/flatten-array.test (hunk 1 above)
Comment (glennj) [reply exercises/practice/flatten-array/flatten-array.test:1]: That's an error. Good eye!

File: exercises/practice/forth/forth.test (hunk 2 above)
Comment (glennj) [reply exercises/practice/forth/forth.test:19]: A lazy cut-n-paste

**Review by kotp — APPROVED** [review]

**Pull request merged**